src/python/pants/option/option_types.py

```diff
@@ -802,6 +803,7 @@ def __new__(
                     """
                 )
             ),
+            default=default, # type: ignore[arg-type]
```

Comment:
Went with the type ignore for now since it does seem like this might be a mypy bug.

Reply:
Thanks for your patience with all this! Mypy is a bit of a challenge sometimes.

Reply:
You are welcome. I'm glad to have typing in python, just eagery looking forward to it becoming better typing over time.